spim_core/spim_base.py

```diff
@@ -112,6 +114,95 @@ def log_to_file(self, filepath: Path, logger: Logger = None,
             log_handler.close()
             logger.removeHandler(log_handler)
 
+    def check_ext_disk_space(self, xtiles, ytiles, ztiles):
+        """Checks ext disk space before scan to see if disk has enough space scan
+
+        :param xtiles: number of x tiles
+        :param ytiles: number of y tiles
+        :param ztiles: number of z tiles
+        """
+        # One tile (tiff) is ~10368 kb
+        est_stack_filesize = self.cfg.bytes_per_image * ztiles
+        est_scan_filesize = est_stack_filesize*xtiles*ytiles
+        if est_scan_filesize >= shutil.disk_usage(self.cfg.ext_storage_dir).free:
+            self.log.error("Not enough space in external directory")
+            raise
+
+    def check_local_disk_space(self, z_tiles):
+        """Checks local disk space before scan to see if disk has enough space for two stacks
+
+           :param z_tiles: number of z tiles
+        """
+
+        est_filesize = self.cfg.bytes_per_image * z_tiles
+        if est_filesize * 2 >= shutil.disk_usage(self.cfg.local_storage_dir).free:
+            self.log.error("Not enough space on disk. Is the recycle bin empty?")
+            raise
+
+    def check_read_write_speeds(self, drive: Path):
```

Comment:
Just so we remember what all of the fio inputs are. I think we should pull --size, --bs, --direct, --numjobs, etc. out into input variables, and document them in this function. We can make the defaults if not specified what the current values are. We should also add a flag for linux, which would change the ioengine to the default linux one which is 'libaio'.